File: ring-android/app/src/main/java/cx/ring/account/SettingItem.java
package cx.ring.account;

import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;
import androidx.annotation.StringRes;

public class SettingItem {
    private final int mTitleRes;
    private final int mImageId;
    private final Runnable mOnClick;

    public SettingItem(@StringRes int titleRes, @DrawableRes int imageId, @NonNull Runnable onClick) {
        mTitleRes = titleRes;
        mImageId = imageId;
        mOnClick = onClick;
    }

    public @StringRes int getTitleRes() {
        return mTitleRes;
    }

    public @DrawableRes int getImageId() {
        return mImageId;
    }

    public void onClick() {
        mOnClick.run();
    }
}

File: jami-android/app/src/main/java/cx/ring/account/SettingItem.kt
package cx.ring.account

import androidx.annotation.StringRes
import androidx.annotation.DrawableRes

class SettingItem(
    @StringRes val titleRes: Int,
    @DrawableRes val imageId: Int,
    private val mOnClick: Runnable
) {
    fun onClick() {
        mOnClick.run()
    }
}